aderyn_core/src/context/browser/siblings.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{
+    ast::*,
+    context::workspace_context::{ASTNode, WorkspaceContext},
+    visitor::ast_visitor::{ASTConstVisitor, Node},
+};
+
+use super::GetImmediateChildren;
+use super::SortNodeReferencesToSequence;
+
+pub trait GetNextSibling {
+    /// Get the next sibling an ASTNode
+    fn next_sibling<'a>(&self, context: &'a WorkspaceContext) -> Option<&'a ASTNode>;
+}
+
+pub trait GetPreviousSibling {
+    /// Get the previous sibling an ASTNode
+    fn previous_sibling<'a>(&self, context: &'a WorkspaceContext) -> Option<&'a ASTNode>;
+}
+
+#[derive(Default)]
+struct NodeIDReceiver {
+    id: Option<NodeID>,
+}
+
+impl ASTConstVisitor for NodeIDReceiver {
+    fn visit_node_id(&mut self, node_id: Option<NodeID>) -> eyre::Result<()> {
+        self.id = node_id;
+        Ok(())
+    }
+}
+
+impl<T: Node + ?Sized> GetNextSibling for T {
+    fn next_sibling<'a>(&self, context: &'a WorkspaceContext) -> Option<&'a ASTNode> {
+        // Setup a Node ID receiver
+        let mut node_id_receiver = NodeIDReceiver::default();
+
+        // Find the ID of the node this method is called upon
+        self.accept_id(&mut node_id_receiver).ok()?;
+        let current_node_id = node_id_receiver.id?;
+
+        let parent = context.get_parent(current_node_id)?;
+        let children = parent.children(context)?;
+        let sorted_children = children.sort_by_line_nos(context)?;
+
+        for i in 0..sorted_children.len() - 1 {
```

Comment:
Is there a more efficient way of doing this loop?

The cost varies based on the number of siblings, and the position of the current sibling. Is there a way we can make this cost the same no matter the position&sibling numbers?

Reply:
I'll check if the IDs in are also sorted when nodes are sorted by src position. If yes, we could do binary search